### ISSUE TITLE
Edge Zone can not register iso 

### ISSUE THREAD

**Issue by hooklee2000** [apache/cloudstack#7284]
##### ISSUE TYPE

 * Bug Report
 

##### COMPONENT NAME
API,  UI,

~~~

~~~

##### CLOUDSTACK VERSION
main ,4.18.0.0 RC1

~~~
We created Edge Zone named edge1,when registered iso , we can not choose the edge1(only all zone), then it rasied error:
Unable to find image store to download template null 

![Uploading image.png…]()

**Comment by hooklee2000** [comment]
![image](https://user-images.githubusercontent.com/52850777/220881890-ee51bacf-feb6-4789-aa36-69d3af2ae984.png)

**Comment by shwstppr** [comment]
Should be fixed with #8142 allowing registering direct-download ISO on edge zones. @hooklee2000 please re-open if the issue still exists
